project/adc_driver/src/main.c

```diff
@@ -0,0 +1,17 @@
+#include "adc.h"
+#include "gpio.h"
+
+int main() {
+  GPIOAddress address = { 0, 0 };
+  GPIOSettings settings = { GPIO_DIR_IN, GPIO_STATE_LOW, GPIO_RES_NONE, GPIO_ALTFN_ANALOG };
+
+  gpio_init();
+  gpio_init_pin(&address, &settings);
+
+  adc_init(ADC_MODE_CONTINUOUS);
+  adc_init_pin(&address, ADC_SAMPLE_RATE_1);
+
+  while (1) {
+    printf("%d mV\n", adc_read(&address, 3000));
```

Comment:
You should be able to just use the ``LOG`` macro for this. You can check ``firmware/libraries/ms-common/test/test_fsm.c`` for some examples on how to use it.